async-nats/src/lib.rs

```diff
@@ -1197,6 +1308,48 @@ impl Stream for Subscriber {
     }
 }
 
+#[derive(Debug)]
+pub struct Errors {
+    receiver: tokio::sync::mpsc::Receiver<ServerError>,
+}
+
+impl Errors {
+    fn new(receiver: tokio::sync::mpsc::Receiver<ServerError>) -> Errors {
+        Errors { receiver }
+    }
+}
+
+impl Stream for Errors {
+    type Item = ServerError;
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        self.receiver.poll_recv(cx)
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum ServerError {
+    AuthorizationViolation,
+    Other(String),
+}
+
+impl ServerError {
+    fn new(error: String) -> ServerError {
+        match error.as_str() {
```

Comment:
Needs a rebase? pretty sure you corrected this and made it lowercase already.

```suggestion
        match error.as_str() {
        
```

Reply:
yep. rebased.